### PR TITLE
[INLONG-8316][Dashboard] Update length limit for partial ID or name field

### LINKED ISSUE #8316: [Improve][Dashboard] Update length limit for partial ID or name field
### Description

Update length limit for partial ID or name field.

### InLong Component

InLong Dashboard

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haifxu**
### Prepare a Pull Request

- Fixes #8316 

### Motivation

Consistent with the Manager: https://github.com/apache/inlong/pull/8283

### Modifications

1. Update length limit for partial ID or name field
2. Modify some validation rules

### Verifying this change

- [x] This change is a trivial rework/code cleanup without any test coverage.

### CODE REVIEW

**Review by dockerzhang — APPROVED** [review]

File: inlong-dashboard/src/plugins/groups/common/GroupDefaultInfo.ts [hunk 1]
```diff
@@ -39,12 +39,12 @@ export class GroupDefaultInfo implements DataWithBackend, RenderRow, RenderList
...
     props: {
-      maxLength: 100,
+      maxLength: 200,
     },
     rules: [
       { required: true },
       {
-        pattern: /^[a-z_0-9]+$/,
+        pattern: /^[a-zA-Z0-9_.-]+$/,
         message: i18n.t('meta.Group.InlongGroupIdRules'),
```
Comment (healchow) [inline inlong-dashboard/src/plugins/groups/common/GroupDefaultInfo.ts:48]: Please update the description of `meta.Group.InlongGroupIdRules` at the same time.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-dashboard/src/plugins/groups/common/GroupDefaultInfo.ts (hunk 1 above)
Comment (haifxu) [reply inlong-dashboard/src/plugins/groups/common/GroupDefaultInfo.ts:48]: Done.

**Review by leezng — APPROVED** [review]

File: inlong-dashboard/src/ui/locales/cn.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
   "basic.SuccessfullyRestart": "重启成功",
   "basic.SuccessfullyStop": "停止成功",
   "meta.Sources.Name": "数据源名称",
+  "meta.Sources.NameRule": "只能包含英文字母、数字、中划线、下划线",
```
Comment (healchow) [inline inlong-dashboard/src/ui/locales/cn.json:25]:
```suggestion
  "meta.Sources.NameRule": "只能包含英文字母、数字、中划线(-)、下划线(_)",
```

File: inlong-dashboard/src/ui/locales/cn.json [hunk 3]
```diff
@@ -362,7 +363,7 @@
   "meta.Sinks.Kudu.FieldDescription": "描述",
   "meta.Sinks.Kudu.buckets": "Buckets",
   "meta.Group.InlongGroupId": "数据流组 ID",
-  "meta.Group.InlongGroupIdRules": "只能包含小写字母、数字、中划线、下划线",
+  "meta.Group.InlongGroupIdRules": "只能包含英文字母、数字、中划线、下划线",
```
Comment (healchow) [inline inlong-dashboard/src/ui/locales/cn.json:366]: **Other Chinese and English descriptions are also suggested to be modified in this way.**

```suggestion
  "meta.Group.InlongGroupIdRules": "只能包含英文字母、数字、点号(.)、中划线(-)、下划线(_)",
```

File: inlong-dashboard/src/ui/locales/cn.json [hunk 4]
```diff
@@ -22,6 +22,7 @@
   "basic.SuccessfullyRestart": "重启成功",
   "basic.SuccessfullyStop": "停止成功",
   "meta.Sources.Name": "数据源名称",
+  "meta.Sources.NameRule": "只能包含英文字母、数字、点号(.)、中划线(-)、下划线(_)",
```
Comment (healchow) [inline inlong-dashboard/src/ui/locales/cn.json:25]: It seems that the source name not support dot(.) =.=!

File: inlong-dashboard/src/ui/locales/en.json [hunk 5]
```diff
@@ -22,6 +22,7 @@
   "basic.SuccessfullyRestart": "Successfully restart",
   "basic.SuccessfullyStop": "Successfully stop",
   "meta.Sources.Name": "Source Name",
+  "meta.Sources.NameRule": "Only English letters, numbers, dots(.), minus(-), and underscores(_)",
```
Comment (healchow) [inline inlong-dashboard/src/ui/locales/en.json:25]: ditto

File: inlong-dashboard/src/plugins/sources/common/SourceDefaultInfo.ts [hunk 6]
```diff
@@ -73,10 +74,16 @@ export class SourceDefaultInfo implements DataWithBackend, RenderRow, RenderList
 
   @FieldDecorator({
     type: 'input',
-    rules: [{ required: true }],
+    rules: [
+      { required: true },
+      {
+        pattern: /^[a-zA-Z0-9_-]*$/,
```
Comment (healchow) [inline inlong-dashboard/src/plugins/sources/common/SourceDefaultInfo.ts:80]:
```suggestion
        pattern: /^[a-zA-Z0-9_.-]*$/,
```

**Review by healchow — APPROVED** [review]

**Pull request merged**